File: eRents.UI/e_rents_mobile/lib/core/models/notification.dart
class Notification {
  final int id;
  final String title;
  final String message;
  final DateTime date;
  final bool isRead;

  Notification({
    required this.id,
    required this.title,
    required this.message,
    required this.date,
    required this.isRead,
  });

  factory Notification.fromJson(Map<String, dynamic> json) {
    return Notification(
      id: json['id'],
      title: json['title'],
      message: json['message'],
      date: DateTime.parse(json['date']),
      isRead: json['isRead'],
    );
  }
}

File: eRents.Domain/Models/Notification.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace eRents.Domain.Models;

public partial class Notification
{
    public int NotificationId { get; set; }

    public int UserId { get; set; }

    [Required]
    public string Title { get; set; } = null!;

    [Required]
    public string Message { get; set; } = null!;

    [Required]
    public string Type { get; set; } = null!; // 'maintenance', 'booking', 'message', 'system'

    public int? ReferenceId { get; set; }  // ID of related entity (bookingId, maintenanceIssueId, etc.)

    public bool IsRead { get; set; } = false;

    public DateTime DateCreated { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public virtual User User { get; set; } = null!;
} 